More failing tests

kevinkhao opened the pull request:
This adds some failing tests

kevinkhao committed:
Merge xlsx refactor

kevinkhao committed:
[TMP] swap

kevinkhao committed:
[FIX] Fix tests after moving to openpyxl

kevinkhao committed:
[WIP] Adding excel tests to cover failure with bad id values

kevinkhao committed:
Add some tests for bad data

kevinkhao committed:
Modify tests to pass before merging

sebastienbeau commented on pattern_import_export/views/pattern_import_export.xml:
```diff
@@ -45,15 +45,18 @@
                 <field name="related_model_id" invisible="1"/>
                 <field name="last_field_id" invisible="1"/>
-                <field name="select_tab_id"
+                <field name="add_select_tab" invisible="1"
```
```suggestion
                <field name="add_select_tab"
```
invisibility should not be forced

sebastienbeau commented on pattern_import_export/models/ir_exports_line.py:
```diff
@@ -2,14 +2,20 @@
 
-    select_tab_id = fields.Many2one("ir.exports.select.tab", string="Select tab")
+    add_select_tab = fields.Boolean()
+    use_tab = fields.Many2one("ir.filters")
```
We should use a better namming, **tab_filter_id** ?

sebastienbeau commented on pattern_import_export/models/ir_exports_line.py:
```diff
@@ -209,3 +230,68 @@ def _get_header(self, use_description=False):
+        result = []
+        for itr, rec in enumerate(self, start=1):
+            if not (
+                rec.related_model_relation_type in ("many2many", "many2one")
```
It will be better to only test the add_select_tab and have a correct value for this field.
When we change the last field we should untick the add_select_tab in some case (we should also have this behaviour for the onchange)

sebastienbeau commented on pattern_import_export_xlsx/models/ir_exports.py:
```diff
@@ -17,30 +17,68 @@ class IrExports(models.Model):
+        Get the actual data and write it row by row on the main sheet
+        """
+        for row, values in enumerate(self._get_data_to_export(records), start=2):
+            for col, header in enumerate(self._get_header(), start=1):
```
maybe we should put the result of self._get_header() into a variable to avoid to call the method for each row

sebastienbeau commented on pattern_import_export_xlsx/tests/test_pattern_import.py:
```diff
@@ -29,20 +29,30 @@ def setUpClass(cls):
         )
+        cls.ir_export_users = cls.env["ir.exports"].create(
+            {
+                "name": "Partner",
```
```suggestion
                "name": "User",
```

kevinkhao committed:
Update pattern_import_export_xlsx/tests/test_pattern_import.py

Co-authored-by: beau sebastien <sebastien.beau@akretion.com>

kevinkhao committed:
[IMP] Modifications according to feedback
- Change "use_tab" to "tab_filter_id"
- Add domain on UI on tab_filter_id
- Small optimization on _populate_main_sheet_rows
- Remove some unused code

sebastienbeau commented on pattern_import_export/views/pattern_import_export.xml:
```diff
@@ -45,18 +45,19 @@
-                <field name="use_tab"
+                <field name="related_model_name" invisible="1"/>
+                <field name="tab_filter_id"
+                       domain="[('model_id', '=', related_model_name)]"
```
instead of adding the field "related_model_name" you can directly filter on "related_model_id"

sebastienbeau replied:
hum in fact model_id is a model name not an id. So it's correct

sebastienbeau reviewed (approved):
LGTM
